src/snowflake/snowpark/table.py

```diff
@@ -267,15 +268,36 @@ class Table(DataFrame):
     with the name of the table in Snowflake. See examples in :meth:`Session.table`.
     """
 
+    @unique
+    class _TableConstructor(Enum):
+        TABLE_INIT = auto()
+        SESSION_TABLE = auto()
+
     def __init__(
         self,
         table_name: str,
         session: Optional["snowflake.snowpark.session.Session"] = None,
+        constructor: _TableConstructor = _TableConstructor.TABLE_INIT,
     ) -> None:
-        # TODO: what do we do if there's no session?
-        stmt = session._ast_batch.assign()
-        stmt.expr.sp_table.table = table_name
-        set_src_position(stmt.expr.sp_table.src)
+        # Begin AST
+        if session is not None:
+            stmt = session._ast_batch.assign()
+            ast = stmt.expr.sp_table
```

Comment:
there's a helper with_src_position(...) that can be used here I believe:
```
ast = with_src_position(stmt.expr.sp_table)
```

Reply:
Done